mmv1/products/edgecontainer/VpnConnection.yaml

```diff
@@ -0,0 +1,151 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: "VpnConnection"
+base_url: "projects/{{project}}/locations/{{location}}/vpnConnections"
+create_url: "projects/{{project}}/locations/{{location}}/vpnConnections?vpnConnectionId={{name}}"
+description: "A VPN connection"
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    "Google Distributed Cloud Edge": "https://cloud.google.com/distributed-cloud/edge/latest/docs"
+  api: "https://cloud.google.com/distributed-cloud/edge/latest/docs/reference/container/rest/v1/projects.locations.vpnConnections"
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: "name"
+    base_url: "{{op_id}}"
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 30
+      update_minutes: 30
+      delete_minutes: 30
+  result: !ruby/object:Api::OpAsync::Result
+    path: "response"
+  status: !ruby/object:Api::OpAsync::Status
+    path: "done"
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: "error"
+    message: "message"
+import_format: ["projects/{{project}}/locations/{{location}}/vpnConnections"]
+autogen_async: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "edgecontainer_vpn_connection"
+    primary_resource_id: "default"
+    skip_test: true
```

Comment:
Skip the default node-pool test as we only have limited machine resources, and the vpn connection is not a critical feature.